frontend/src/components/ViewActivities/activity.js

```diff
@@ -1,25 +1,106 @@
 import React from 'react';
-import Card from 'react-bootstrap/Card';
 import * as time from '../Utils/time.js';
 import * as DB from '../../constants/database.js'
 import '../../styles/activities.css';
-import Accordion from 'react-bootstrap/Accordion';
+import { getField } from './activityfns.js';
+import { Accordion, Button, Card, Col, Form, Row } from 'react-bootstrap';
 
+/**
+ * Returns a dropdown of all the timezones.
+ */
+function timezonePicker() {
+  // TODO: Make this dropdown. (#51)
+  return <div></div>
+}
+
+function submitEdit(k) {
+  console.log(k);
+}
+
+/**
+ * A single activity. 
+ * 
+ * @param {Object} props This component expects the following props:
+ * - `activity` {Object} The activity to display.
+ */
 class Activity extends React.Component {
+  /** {@inheritdoc} */
+  constructor(props) {
+    super(props);
+
+    this.state = { editing: false };
+
+    // Bind state users/modifiers to `this`.
+    this.setEditActivity = this.setEditActivity.bind(this);
+    this.finishEditActivity = this.finishEditActivity.bind(this);
+    this.displayCard = this.displayCard.bind(this);
```

Comment:
According to the [docs](https://reactjs.org/docs/handling-events.html) I think you do not need to have these lines because you are using the "experimental public class fields syntax for your listeners.